### PR TITLE
PLFM-6554: Range and dispatcher workers

### PR DESCRIPTION
**Pull request by marcomarasca**

### CODE REVIEW

File: lib/jdomodels/src/test/java/org/sagebionetworks/repo/model/dbo/dao/files/FilesScannerStatusDaoTest.java [hunk 1]
```diff
@@ -81,60 +93,140 @@ public void testGetWithNonExisting() {
...
-	public void testGetLatestWithEmpty() {
+	public void testExistsWithinLastWithOld() {
+		int numberOfDays = 5;
 		
-		// Call under test
-		Optional<DBOFilesScannerStatus> result = dao.getLatest();
+		long jobId = dao.create().getId();
 		
-		assertFalse(result.isPresent());
+		dao.setUpdatedOn(jobId, numberOfDays);
```
Comment (john-hill) [inline lib/jdomodels/src/test/java/org/sagebionetworks/repo/model/dbo/dao/files/FilesScannerStatusDaoTest.java]: reduceUpdatedOnByNumberOfDays()

File: services/repository-managers/src/test/java/org/sagebionetworks/repo/manager/file/FileHAndleAssociationScannerNotifierUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+		
+		when(mockConfig.getQueueName(any())).thenReturn("QueueName");
+		when(mockSqsClient.getQueueUrl(anyString())).thenReturn(mockQueueUrlResult);
+		when(mockQueueUrlResult.getQueueUrl()).thenReturn(queueUrl);
+		
+		notifier.configureQueue(mockConfig);
+		
+		verify(mockConfig).getQueueName("FILE_HANDLE_SCAN_REQUEST");
+		verify(mockSqsClient).getQueueUrl("QueueName");
+	}
```
Comment (john-hill) [inline services/repository-managers/src/test/java/org/sagebionetworks/repo/manager/file/FileHAndleAssociationScannerNotifierUnitTest.java:71]: need a test for reading and writing an actual json message

File: services/workers/src/main/resources/main-scheduler-spb.xml [hunk 3]
```diff
@@ -123,6 +124,7 @@
 		<ref bean="accessApprovalReminderNotificationWorkerTrigger" />
 		<ref bean="schemaValidationTrigger" />
 		<ref bean="multipartCleanupWorkerTrigger" />
+		<ref bean="fileHandleAssociationScanRangeWorkerTrigger" />
```
Comment (john-hill) [inline services/workers/src/main/resources/main-scheduler-spb.xml:128]: Missing the dispatcher worker trigger.

File: services/workers/src/main/java/org/sagebionetworks/file/worker/FileHandleAssociationScanRangeWorker.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+	public void run(ProgressCallback progressCallback, Message message) throws RecoverableMessageException, Exception {
+		
+		final FileHandleAssociationScanRangeRequest request;
+		
+		try {
+			request = notifier.fromSqsMessage(message);
+		} catch (Throwable e) {
+			LOG.error("Could not process SQS message \n" + message.getBody() + "\n" + e.getMessage(), e);
+			logWorkerCountMetric(METRIC_PARSE_MESSAGE_ERROR_COUNT);
+			throw e;
```
Comment (john-hill) [inline services/workers/src/main/java/org/sagebionetworks/file/worker/FileHandleAssociationScanRangeWorker.java]: handle or throw but not both.

File: services/workers/src/main/java/org/sagebionetworks/file/worker/FileHandleAssociationScanRangeWorker.java [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+		} catch (NotFoundException e) {
+			LOG.warn(e.getMessage(), e);
+		} catch (RecoverableException e) {
+			LOG.warn(requestToString(request) + " RETRYING: " + e.getMessage(), e);
+			logWorkerCountMetric(METRIC_JOB_RETRY_COUNT);
+			throw new RecoverableMessageException(e.getMessage(), e);
+		} catch (Throwable e) {
+			LOG.error(requestToString(request) + " FAILED: " + e.getMessage(), e);
+			logWorkerCountMetric(METRIC_JOB_FAILED_COUNT);
+			throw e;
```
Comment (john-hill) [inline services/workers/src/main/java/org/sagebionetworks/file/worker/FileHandleAssociationScanRangeWorker.java]: no throw

**Review by john-hill — APPROVED** [review]

**Comment by marcomarasca** [comment]
Note that we cannot merge this until we deploy from the develop branch of the stack-builder where we defined the new bucket, stream and queue. At the moment we are deploying from a branch because the develop one includes the new VPC changes.

**Review by john-hill — APPROVED** [review]

**Pull request merged**